parse_opcodes

```diff
@@ -631,6 +640,12 @@ def make_rust():
       lo = arglut[arg][1]
       hi = arglut[arg][0]
       print("        {}: (raw >> {}) & 0x{:x},".format(arg.lower(), lo, (1<<(hi-lo+1))-1))
+      # Parsing for imm6 in correct order, hard-encoded
+      if arg=="imm6hi":
+        imm6lo_lo = arglut["imm6lo"][1]
+        imm6lo_hi = arglut["imm6lo"][0]
+        print("        {}: ((raw >> {}) & 0x{:x}) << 1 | (raw >> {}) & 0x{:x},"\
+        .format("imm6", imm6lo_lo, (1<<(imm6lo_hi-imm6lo_lo+1))-1, lo, (1<<(hi-lo+1))-1))
```

Comment:
I think you can reuse the structure that `bimm12` provides. Sorry, I also just saw that while reviewing, but it seems other immediates have the same issue and there is already a standard way to handle it. Check out `bimm12`'s definition of the shuffling: https://github.com/oakentling/riscv-opcodes/blob/fixes/parse_opcodes#L68 and this function to generate the extraction: https://github.com/oakentling/riscv-opcodes/blob/fixes/parse_opcodes#L481